zkevm-circuits/src/exp_circuit.rs

```diff
@@ -382,37 +401,137 @@ impl<F: Field> ExpCircuitConfig<F> {
+                region.name_column(|| "exp_table.identifier", self.exp_table.identifier);
+                region.name_column(|| "exp_table.is_last", self.exp_table.is_last);
+                region.name_column(|| "exp_table.base_limb", self.exp_table.base_limb);
+                region.name_column(|| "exp_table.exponent_lo_hi", self.exp_table.exponent_lo_hi);
+                region.name_column(
+                    || "exp_table.exponentiation_lo_hi",
+                    self.exp_table.exponentiation_lo_hi,
+                );
+                // region.name_column(|| "q_usable", self.q_usable);
+                region.name_column(|| "mul_gadget.col0", self.mul_gadget.col0);
+                region.name_column(|| "mul_gadget.col1", self.mul_gadget.col1);
+                region.name_column(|| "mul_gadget.col2", self.mul_gadget.col2);
+                region.name_column(|| "mul_gadget.col3", self.mul_gadget.col3);
+                region.name_column(|| "mul_gadget.col4", self.mul_gadget.col4);
+                // region.name_column(|| "mul_gadget.overflow", self.mul_gadget.overflow);
+                region.name_column(|| "parity_check.col0", self.parity_check.col0);
+                region.name_column(|| "parity_check.col1", self.parity_check.col1);
+                region.name_column(|| "parity_check.col2", self.parity_check.col2);
+                region.name_column(|| "parity_check.col3", self.parity_check.col3);
+                region.name_column(|| "parity_check.col4", self.parity_check.col4);
+                // region.name_column(|| "parity_check.overflow", self.parity_check.overflow);
+                debug_assert!(
+                    max_exp_rows >= self.minimum_rows,
+                    "max_exp_rows: {}, minimum_rows: {}",
+                    max_exp_rows,
+                    self.minimum_rows,
+                );
+                let mut offset = 0;
+                for exp_event in exp_events.iter() {
+                    self.assign_exp_event(
+                        &mut region,
+                        &mut offset,
+                        exp_event,
+                        &mut mul_chip,
+                        &mut parity_check_chip,
+                    )?;
+                }
+                assert!(offset <= max_exp_rows - UNUSABLE_EXP_ROWS);
+                dbg!(max_exp_rows);
+                dbg!(self.minimum_rows);
+                // Fill the rest of the circuit with valid rows to achieve a constant assignment
+                // to the q_usable fixed column.
+                let (one, two) = (U256::one(), U256::from(2u64));
+                let pad_exp_event = ExpEvent {
+                    identifier: 0,
+                    base: one,
+                    exponent: two,
+                    exponentiation: one,
+                    steps: vec![ExpStep {
+                        a: one,
+                        b: one,
+                        d: one,
+                    }],
+                };
```

Comment:
We already have `ExpEvent::default` for this. So you don't need to construct it.

Reply:
Even if you assign `1 ^ 1 == 1`, the mul gadget will take up 7 rows for that step. So instead, we should use `2 ^ 2 == 4` as this is the base case for exponentiation trace.